atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -500,8 +509,10 @@ public Map<byte[], Iterator<Map.Entry<Cell, byte[]>>> getRowsColumnRangeIterator
                     return scopeToTransaction(entryIterator);
                 }));
 
+        hasReadsThatRequireImmutableTimestampLockValidationAtCommitRound = true;
```

Comment:
question for reviewer: In theory we could set these back to `false` if we actually did a check in place, so we don't need to do it again at commit round. But that would require us to `synchronize` access to `validatePreCommitRequirementsOnReadIfNecessary`. Is that an acceptable trade off in a world where we can have many threads executing gets and puts for the same transaction?

Reply:
Interesting! Conceptually we could just have a concurrent hash-set of outstanding request IDs and check it's empty at the end to avoid some of the contention during execution?

Probably not for the scope of this PR unless you feel strongly about it.

Reply:
Ah, nice suggestion! Made this small PR giving it a go: https://github.com/palantir/atlasdb/pull/6628

Let me know what you prefer!